src/YAMLValidator/Validator.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Json.Schema;
+using Yaml2JsonNode;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Micorosoft.PowerPlatform.PowerApps.Persistence;
+namespace Microsoft.PowerPlatform.PowerApps.Persistence;
+
+internal sealed class Validator
+{
+    private readonly EvaluationOptions _verbosityOptions;
+    private readonly JsonSerializerOptions _serializerOptions;
+
+
+    public Validator(EvaluationOptions options, JsonSerializerOptions resultSerializeOptions)
+    {
+        // to do: add verbosity flag and allow users to choose verbosity of evaluation
+        _verbosityOptions = options;
+        _serializerOptions = resultSerializeOptions;
+
+    }
+
+    public YamlValidatorResults Validate(JsonSchema schema, string yamlFileData)
+    {
+        var yamlStream = YamlValidatorUtility.MakeYamlStream(yamlFileData);
```

Comment:
if `YamlStream` implements `IDisposable`, please add a `using` statement here

Reply:
[It doesn't implement the IDisposable interface from the source code it seems](https://github.com/aaubry/YamlDotNet/blob/master/YamlDotNet/RepresentationModel/YamlStream.cs.